src/flowchem/devices/knauer/knauer_autosampler_component.py

```diff
@@ -0,0 +1,315 @@
+    """
+
+    hw_device: KnauerAutosampler  # for typing's sake
+
+    def __init__(self, name: str, hw_device: KnauerAutosampler) -> None:
+        """Initialize component."""
+        super().__init__(name, hw_device)
+
+    async def infuse(self, rate: str = None, volume: str = None) -> bool:  # type: ignore
+        """
+        Dispense with built in syringe.
+        Args:
+            volume: volume to dispense in mL
+
+        Returns: None
+        """
+        if volume is None:
+            volume = "0 mL"
+            logger.warning(f"the volume to infuse is not provided. set to 0 ml")
+        parsed_volume = ureg.Quantity(volume)
+        success = await self.hw_device.dispense(volume=parsed_volume.m_as("mL"))
+        if success:
+            logger.info(f"Syringe pump successfully infused {volume} ml")
+
+    async def withdraw(self, rate: str = None, volume: str = None) -> bool:  # type: ignore
+        """
+        Aspirate with built in syringe.
+        Args:
+            volume: volume to aspirate in mL
+
+        Returns: None
+        """
+        if volume is None:
+            volume = self.hw_device.syringe_volume
+            logger.warning(f"the volume to withdraw is not provided. set to {self.hw_device.syringe_volume}")
+        parsed_volume = ureg.Quantity(volume)
+        success = await self.hw_device.aspirate(volume=parsed_volume.m_as("mL"))
+        if success:
+            logger.info(f"Syringe pump successfully withdrew {volume} ml")
+
+    #ToDo
+    async def stop(self):  # type: ignore
+        """Stop pumping."""
+        ...
+
+    # ToDo
+    async def is_pumping(self) -> bool:  # type: ignore
+        """Is pump running?"""
+        ...
+
+    @staticmethod
+    def is_withdrawing_capable() -> bool:  # type: ignore
+        """Can the pump reverse its normal flow direction?"""
+        return True
+
+
+class AutosamplerSyringeValve(FourPortDistributionValve):
+    """
+    Control a Knauer Autosampler 4-Port Distribution syringe Valve.
+
+    Attributes:
+        hw_device (KnauerAutosampler): The hardware device for the 4-Port Distribution syringe Valve.
+    """
+
+    hw_device: KnauerAutosampler  # for typing's sake
+    identifier = "syringe_valve"
+
+    def __init__(self, name: str, hw_device: KnauerAutosampler) -> None:
+        super().__init__(name, hw_device)
+        self.add_api_route("/monitor_position", self.get_monitor_position, methods=["GET"])
+        self.add_api_route("/monitor_position", self.set_monitor_position, methods=["PUT"])
+
+    def _change_connections(self, raw_position: str | int, reverse: bool = False):
+        """
+        Change connections based on the valve's raw position.
+
+        Args:
+            raw_position (int): The raw position of the valve.
+            reverse (bool): Whether to reverse the mapping.
+
+        Returns:
+            int: The mapped position.
+        """
+        position_mapping = {0: "NEEDLE", 1: "WASH", 2: "WASH_PORT2", 3: "WASTE"}
```

Comment:
Now I'm thinking that this naming (WASH, NEEDLE...) corresponds to the sugar platform. I should chang it to the one in the AS guide no? And then in the sugar platform rename it if necessary.

Reply:
This should inherit from the 4-Portdistribution valve and be implemented like any other valve. You just need to define the appropriate offset and reorient the valve. On top, you can add this mapping for easy of use

Reply:
kind of - all valves shopuld strictly use valve logic. any mapping that is not a mathematical function should just happen for the monitor_function which is just for easy user access